tsdb/chunkenc/histogram_meta.go

```diff
@@ -558,11 +558,12 @@ func counterResetHint(crh CounterResetHeader, numRead uint16) histogram.CounterR
 		// In a counter histogram chunk, there will not be any counter
 		// resets after the first histogram.
 		return histogram.NotCounterReset
-	case crh == CounterReset:
-		// If the chunk was started because of a counter reset, we can
-		// safely return that hint. This histogram always has to be
-		// treated as a counter reset.
-		return histogram.CounterReset
+		// TODO: explain
+	//case crh == CounterReset:
+	// If the chunk was started because of a counter reset, we can
+	// safely return that hint. This histogram always has to be
+	// treated as a counter reset.
+	//return histogram.CounterReset
 	default:
```

Comment:
The comment below needs a rework based on our current understanding. I also think my TODO below (line 575) is pretty much obsolete.

Reply:
Reworked the comment in https://github.com/prometheus/prometheus/pull/15343/commits/8311877dfc99e1ca9ab56a9a13dad8db05652041

Reply:
Also fixed a bit in https://github.com/prometheus/prometheus/pull/15343/commits/01717673255f7ad573200d3f3cb06f4807eab7b2